### PR TITLE
Lower disk size for vSphere generated e2e tests

### PR DESCRIPTION
**Pull request by kron4eg**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
To decrease storage pressure when multiple vsphere e2e tests are launched

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xmudrii** [comment]
/override pull-kubeone-e2e-vsphere-default-install-containerd-v1.23.8
There are known problems with our vSphere infra, so it's expected that tests are failing.

File: examples/terraform/vsphere/variables.tf [hunk 1]
```diff
@@ -157,13 +157,25 @@ variable "control_plane_memory" {
...
   type        = number
 
 }
 
+variable "worker_num_cpus" {
+  default     = 2
+  description = "number of cpus of each workers node"
+  type        = number
+}
+
```
Comment (ahmedwaleedmalik) [inline examples/terraform/vsphere/variables.tf:178]: Just a nit; can we please implement this for `vsphere-flatcar` as well? I know it's benign but I'd love to keep it somewhat consistent.

Comment (xmudrii) [reply examples/terraform/vsphere/variables.tf]: That's a good point. Fixed.

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve 🛸

**Pull request merged**